zero/gkyl_array_average.h

```diff
@@ -0,0 +1,76 @@
+#pragma once
+
+#include <gkyl_array.h>
+#include <gkyl_basis.h>
+#include <gkyl_range.h>
+#include <gkyl_rect_grid.h>
+
+// Object type
+typedef struct gkyl_array_average gkyl_array_average;
+
+// Input of the new routine is packaged as a struct
+typedef struct gkyl_array_average_inp gkyl_array_average_inp;
+/**
+ * The input structure contains the parameters required to perform a weighted array average:
+ * 
+ * @param grid          Pointer to the computational grid, used to compute the surface element.
+ * @param basis         Total basis, describes the full dimensionality (ndim), polynomial type, and order.
+ * @param basis_avg     Subset basis, describes the reduced dimensionality, polynomial type, and order for the output.
+ * @param local         Full range of the input array, covering all dimensions of the total basis.
+ * @param local_avg     Reduced range of the output array, covering only the non-averaged dimensions.
+ * @param local_avg_ext Extended reduced range of the output array, used only to define the integrated weight.
+ * @param weight        Pointer to the array containing weight for the averaging process. (set it to NULL for integral)
+ * @param avg_dim       Flag array to set which dimension is averaged
+ * @param use_gpu       Boolean flag indicating whether the computation should be performed on a GPU.
```

Comment:
I used to use spaces like these and got heavily criticized for years. The code doesn't use them, so please remove.